### PR TITLE
Android build fixes

### PR DESCRIPTION
**Pull request by Rugvip**
This along with a libffi-version fix, twi_'s patches seems to fix everything but the libgnustl issue

### CODE REVIEW

**Comment by nirbheek** [comment]
The first patch is fine, but you should use shell.replace() on the file instead of read() + re.sub() [see, for example, openh264.recipe on how to use it]. The second patch is wrong since post_install is still needed, as discussed on IRC.

**Comment by Rugvip** [comment]
`shell.replace` doesn't work with regexp, which is why I had to go with that solution.

**Comment by nirbheek** [comment]
Hum, I thought it supported globbing, but it seems it doesn't. Okay, please just send that patch and I'll merge it. :)

**Comment by Rugvip** [comment]
With these two patches I finally got it to build and run.

I did run into some other minor issues as well, missing libsoup and json-glib dependencies, and a missing version_id file.

File: recipes/openwebrtc.recipe [hunk 1]
```diff
@@ -75,7 +76,9 @@ class Recipe(recipe.Recipe):
                         self.config.target_arch)
                 android_sdk = os.path.join(self.config.build_tools_prefix,
                                            'opt', 'android-sdk')
-                self.configure_options += '--enable-owr-java --enable-shared \
+                self.configure_options += '--enable-shared  \
+                    --enable-owr-java \
+                    --with-jardir=' + os.path.join(self.config.prefix, 'lib', 'jni') + ' \
```
Comment (nirbheek) [inline recipes/openwebrtc.recipe]: Isn't this the default already?

configure.ac:
AC_ARG_WITH(jardir,
            AS_HELP_STRING([--with-jardir],
                           [jar file installation path [LIBDIR/jni]]),
            jardir=${withval}, jardir=${libdir}/jni)

**Comment by nirbheek** [comment]
I'm curious about the minor issues you faced; especially the libsoup and json-glib dependency problem. Those should only be used by the tests (and I've added a patch to Cerbero that ensures that they're only checked for when tests are enabled). Could you please expand on that, and maybe file bugs for those?

As for the patch itself, besides the one question I have, could you please add a comment as to why the shared-library regex sub is being done for future reference? Besides this, both patches look great. :)

File: recipes/openwebrtc.recipe (hunk 1 above)
Comment (Rugvip) [inline recipes/openwebrtc.recipe]: True, it didn't seem like any jars were generated though. I'll double check if that was the case and probably fix it in owr instead.

Comment (Rugvip) [inline recipes/openwebrtc.recipe]: Should be fixed in EricssonResearch/openwebrtc#238

**Comment by Rugvip** [comment]
Iirc the openwebrtc config complained about missing libsoup and json-glib, to be able to continue I just built those with cerbero.
I guess the problem should be easily reproducible by cleaning those two libraries and trying to build owr.

I added a comment and removed the jardir patch

**Comment by nirbheek** [comment]
I've fixed the libsoup/json-glib dependency issue (missing whitespace in configure options). Thanks!

**Pull request merged**